File: Client/src/DataPacket.cpp
/*
 * DataPacket.cpp
 *
 *  Created on: Jan 17, 2017
 *      Author: roima
 */

#include "../include/Packet.h"

DataPacket::DataPacket(char* data,short PacketSize,short BlockNum):Packet(3),data(data),PacketSize(PacketSize),BlockNum(BlockNum) {
}
DataPacket::DataPacket(const DataPacket &obj):Packet(3),data(),PacketSize(obj.PacketSize),BlockNum(obj.BlockNum){
    
}
DataPacket& DataPacket::operator=(const DataPacket &obj){
    return *this;
    
}
short DataPacket::getPacketSize(){
	return PacketSize;
}
short DataPacket::getBlockNum(){
	return BlockNum;
}
char* DataPacket::getData(){
	return data;
}
DataPacket::~DataPacket() {

}

File: Server/src/main/java/bgu/spl171/net/api/bidi/DataPacket.java
package bgu.spl171.net.api.bidi;

public class DataPacket extends Packet{

	private short PacketSize;
	private short BlockNum;
	private byte[] data;
	
	public DataPacket(short Packetsize, short BlockNum, byte[]data){
		super.opCode=3;
		this.PacketSize=Packetsize;
		this.BlockNum=BlockNum;
		this.data=data;
	}

	public short getBlockNum(){
		return BlockNum;
	}
	
	public short getPacketSize(){
		return PacketSize;
	}
	
	public byte[] getData(){
		return data;
	}
}
